### ISSUE TITLE
Suggestion: Adding RFC6186

### ISSUE THREAD

**Issue by axoroll7** [wdes/mail-autodiscover-autoconfig#2]
My suggestion: Generating a DNS zone file to include the email client autoconfiguration specified in [RFC6186](https://datatracker.ietf.org/doc/html/rfc6186), or a step-by-step tutorial.
(The update [RFC8314](https://datatracker.ietf.org/doc/html/rfc8314) add "_submissions._tcp" for SMTP over implicit TLS)

Examples:
```text
_imap._tcp SRV 0 1 143 imap.example.com.
_imaps._tcp SRV 0 1 993 imap.example.com.
_submission._tcp SRV 0 1 587 mail.example.com.
_submissions._tcp SRV 0 1 465 mail.example.com.
```

When the "s" is present, implicit TLS is utilized (SSL/TLS), and when it is absent, opportunist TLS is utilized (Plain or STARTTLS).

Edit: gmail.com is a real world example, server-side. I do not have a client-side example for the moment.

**Comment by williamdes** [comment]
Sure !
I can provide a route for this, it will help importing the TXT into some DNS providers, like CloudFlare for example
There is more DNS fields that could be included in it

| --                             | --    | --                                                             |
|--------------------------------|-------|----------------------------------------------------------------|
| example.com                    | TXT   | "mailconf=https://autoconfig.example.com/mail/config-v1.1.xml" |
| autoconfig.example.com         | CNAME | autoconfig-host.example.com                                    |
| autodiscover.example.com       | CNAME | autodiscover-host.example.com                                  |
| _autodiscover._TCP.example.com | SRV   | autodiscover.example.com                                       |
| _pop3._TCP.example.com         | SRV   | pop.mails.example.com                                          |
| _pop3s._TCP.example.com        | SRV   | pop.mails.example.com                                          |

**Comment by williamdes** [comment]
I added some code to make a DNS text file that downloads itself. I need to test it a bit but it should be alright. Will make a release for this soon

**Comment by williamdes** [comment]
I just published [v1.5.0](https://github.com/wdes/mail-autodiscover-autoconfig/releases/tag/v1.5.0) 🎉
